packages/flutter_test/lib/src/window.dart

```diff
@@ -0,0 +1,303 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data' show ByteData;
+import 'dart:ui' hide window;
+
+import 'package:meta/meta.dart';
+
+/// [Window] that wraps another [Window] and allows faking of some properties
+/// for testing purposes.
+///
+/// Tests for certain widgets, e.g., [MaterialApp], might require faking certain
+/// properties of a [Window]. [TestWindow] facilitates the faking of these properties
+/// by overidding the properties of a real [Window] with desired fake values. The
+/// binding used within tests, [TestWidgetsFlutterBinding], contains a [TestWindow]
+/// that is used by all tests.
+///
+/// A test can utilize a [TestWindow] in the following way:
+///
+/// ```dart
+/// testWidgets('your test name here', (WidgetTester tester) async {
+///   // Retrieve the TestWidgetsFlutterBinding.
+///   final WidgetsBinding binding = tester.binding;
```

Comment:
Why not just

`final TestWidgetsFlutterBinding testBinding = tester.binding`

? My understanding is that dart automatically adds the assert for these kinds of upcasts.

Reply:
Yeah, I think that should work.  Changed.